tests/src/test/scala/ch/renku/acceptancetests/workflows/Login.scala

```diff
@@ -4,23 +4,135 @@ import ch.renku.acceptancetests.pages._
 import ch.renku.acceptancetests.tooling.AcceptanceSpec
 import ch.renku.acceptancetests.workflows.LoginType._
 
+import org.openqa.selenium.StaleElementReferenceException
+
 import scala.concurrent.duration._
 import scala.language.postfixOps
 
 trait Login {
   self: AcceptanceSpec =>
 
   def logIntoRenku: LoginType = {
-    Given("User is logged in on the Welcome page")
+    Given("user is not logged in")
     go to LandingPage sleep (1 second)
     verify browserAt LandingPage
-    click on LandingPage.loginButton
+
+    When("user clicks on the Login button")
+    try {
+      click on LandingPage.loginButton
+    } catch {
```

Comment:
As it looks the `StaleElementReferenceException` occurs on the click operation, could we maybe build our own grammar for clicking a button which would do the re-try on that exception internally? I know, there's is this `click` object on the `WebBrowser` but it could be quite useful to have that mechanism in place for all the clicks everywhere. I'm not sure now how difficult it will be to do that but maybe it's worth trying out...

Reply:
The other thing is that maybe it would be good to add some tiny delay before we try to click again. Otherwise, it just shoots like a machine gun :)

Reply:
1. I think that is a good suggestion, but I would wait until we identify more places where this behavior is happening. Right now, I have only seen in on the login, so I'm ok with have a custom solution in this one location until we need a general solution.

2. Good point. Done.

Reply:
You know it better but isn't the `StaleElementReferenceException` being thrown here because we don't have a sleep after the previous step which is `verify browserAt LandingPage`?  I'm just wondering if what could be happening here is that the browser has already the `LandingPage` URL (so the `verify browserAt LandingPage` passes ) but the content is still being loaded while we try to click the LoginButton.

Reply:
Yes, I think there is a possibility that sleeping longer will fix this problem as well...

Reply:
So which option do you reckon would be more robust? From the code simplicity extending the timout seems to be better but to be honest I'm not so sure...

Reply:
I'm increased the timeout. This seem sufficient.